File: serde-generate/runtime/swift/Bincode/BincodeSerializer.swift
//  Copyright © Diem Association. All rights reserved.

import Foundation

public class BincodeSerializer : BinarySerializer {
  
  init() {
    super.init(maxContainerDepth: Int64.max)
  }
  
  public override func serialize_len(value: Int64) {
    output.write(data: Data(fromArray: [value]))
  }
  
  public override func serialize_variant_index(value: Int) {
    output.write(data: Data(fromArray: [value]))
  }
  
  public override func sort_map_entries(offsets: [Int]) {
    // Not required by the format.
  }
}

File: serde-generate/runtime/dart/bincode/bincode_serializer.dart
// Copyright (c) Facebook, Inc. and its affiliates
// SPDX-License-Identifier: MIT OR Apache-2.0
part of bincode;

class BincodeSerializer extends BinarySerializer {
  @override
  void serializeLength(int value) {
    serializeUint64(value);
  }

  @override
  void serializeVariantIndex(int value) {
    serializeUint32(value);
  }

  void sortMapEntries(Int32List offsets) {
    // Not required by the format.
  }
}
